File: scripts/rename-content-files.js
import fs from 'fs';
import path from 'path';

const contentDirs = [
  '/vercel/share/v0-project/content/lessons',
  '/vercel/share/v0-project/content/videos'
];

function removeNumberPrefix(filename) {
  // Remove leading numbers and dash: "0001-" or "0110-" becomes ""
  return filename.replace(/^\d+-/, '');
}

function processDirectory(dir) {
  console.log(`\n📂 Processing: ${dir}`);
  
  const files = fs.readdirSync(dir);
  
  // Filter out non-files and create mapping of old -> new names
  const fileMap = files
    .filter(file => {
      const fullPath = path.join(dir, file);
      return fs.statSync(fullPath).isFile();
    })
    .map(file => ({
      oldName: file,
      newName: removeNumberPrefix(file)
    }))
    .sort((a, b) => a.newName.localeCompare(b.newName));

  // Rename files
  let renamed = 0;
  for (const { oldName, newName } of fileMap) {
    if (oldName === newName) {
      console.log(`  ⏭️  ${oldName} (no change needed)`);
      continue;
    }
    
    const oldPath = path.join(dir, oldName);
    const newPath = path.join(dir, newName);
    
    try {
      fs.renameSync(oldPath, newPath);
      console.log(`  ✅ ${oldName} → ${newName}`);
      renamed++;
    } catch (error) {
      console.error(`  ❌ Failed to rename ${oldName}: ${error.message}`);
    }
  }
  
  console.log(`   Total renamed: ${renamed}/${fileMap.length}`);
}

// Process both directories
for (const dir of contentDirs) {
  if (fs.existsSync(dir)) {
    processDirectory(dir);
  } else {
    console.error(`⚠️  Directory not found: ${dir}`);
  }
}

console.log('\n✅ All content files have been reorganized alphabetically!');

File: scripts/rename-content-files.py
#!/usr/bin/env python3

import os
import re
from pathlib import Path

project_root = Path('/vercel/share/v0-project')
content_dirs = [
    project_root / 'content' / 'lessons',
    project_root / 'content' / 'videos'
]

def remove_number_prefix(filename):
    """Remove leading numbers and dash from filename"""
    return re.sub(r'^\d+-', '', filename)

def process_directory(dir_path):
    print(f'\n📂 Processing: {dir_path}')
    
    if not dir_path.exists():
        print(f'⚠️  Directory not found: {dir_path}')
        return
    
    files = [f for f in dir_path.iterdir() if f.is_file() and f.suffix == '.md']
    print(f'   Found {len(files)} files')
    
    # Create mapping of old -> new names
    file_map = []
    for old_file in files:
        new_name = remove_number_prefix(old_file.name)
        file_map.append((old_file, old_file.parent / new_name))
    
    # Sort by new name
    file_map.sort(key=lambda x: x[1].name.lower())
    
    # Rename files
    renamed = 0
    for old_path, new_path in file_map:
        if old_path == new_path:
            print(f'  ⏭️  {old_path.name} (no change)')
            continue
        
        try:
            old_path.rename(new_path)
            print(f'  ✅ {old_path.name} → {new_path.name}')
            renamed += 1
        except Exception as e:
            print(f'  ❌ Failed: {e}')
    
    print(f'   Renamed: {renamed}/{len(file_map)}')

for dir_path in content_dirs:
    process_directory(dir_path)

print('\n✅ All content files have been reorganized alphabetically!')
